### PR TITLE
Refactor based on discussions in #1650

### PR DESCRIPTION
**Pull request by vaikas**
Signed-off-by: Ville Aikas <vaikas@chainguard.dev>

#### Summary
Refactor code based on discussions in #1650 as followups. 

#### Ticket Link

Fixes

#### Release Note

```release-note

```

### CODE REVIEW

**Comment by vaikas** [comment]
@mattmoor @hectorj2f 
Addressing your PR feedback, ptal :)

File: pkg/cosign/kubernetes/webhook/validator.go [hunk 1]
```diff
@@ -241,39 +245,56 @@ func validatePolicies(ctx context.Context, ref name.Reference, defaultKC authn.K
...
+	// If none of the Authorities for a given policy pass the checks, gather
+	// the errors here. If one passes, do not return the errors.
+	authorityErrors := []error{}
+	for _, authority := range authorities {
+		logging.FromContext(ctx).Debugf("Checking Authority: %+v", authority)
+		// TODO(vaikas): We currently only use the defaultKC, we have to look
+		// at authority.Sources to determine additional information for the
+		// WithRemoteOptions below, at least the 'TargetRepository'
+		// https://github.com/sigstore/cosign/issues/1651
+		opts := ociremote.WithRemoteOptions(remote.WithAuthFromKeychain(kc))
```
Comment (mattmoor) [inline pkg/cosign/kubernetes/webhook/validator.go:272]: Should we also include the `ociremote.Option` argument from above?

**Review by mattmoor — APPROVED** [review]
My one comment isn't new, but we should probably still sort it out.

**Review by hectorj2f — APPROVED** [review]
Lgtm

File: pkg/cosign/kubernetes/webhook/validator.go (hunk 1 above)
Comment (vaikas) [reply pkg/cosign/kubernetes/webhook/validator.go]: Sure, it's never currently used, so just left it in the TODO, but done :)

**Pull request merged**